src/materialized/bin/materialized.rs

```diff
@@ -158,13 +158,24 @@ fn run() -> Result<(), failure::Error> {
     let gather_metrics = !popts.opt_present("no-prometheus");
     let listen_addr = popts.opt_get("listen-addr")?;
 
-    if cfg!(debug_assertions) && !popts.opt_present("dev") && !ore::env::is_var_truthy("MZ_DEV") {
+    let debug_opt_in =
+        cfg!(debug_assertions) && (popts.opt_present("dev") || ore::env::is_var_truthy("MZ_DEV"));
+
+    if cfg!(debug_assertions) && !debug_opt_in {
         bail!(
             "refusing to run dev (unoptimized) binary without explicit opt-in\n\
              hint: Pass the '--dev' option or set the MZ_DEV environment variable to opt in.\n\
              hint: Or perhaps you meant to use a release binary?"
         );
     }
+    if !popts.opt_present("threads") && !debug_opt_in {
+        bail!(
+            "missing required number of worker threads '--threads'\n\
```

Comment:
we might want to say something more explicit about it being required for release builds?

e.g. "missing required number of worker threads for release builds '--threads'"